zingolib/src/lightclient/describe.rs

```diff
@@ -375,104 +396,113 @@ impl LightClient {
+    fn record_value_transfers(
         summaries: &mut Vec<ValueTransfer>,
         txid: TxId,
-        transaction_md: &TransactionRecord,
-    ) {
+        transaction_record: &TransactionRecord,
+        transaction_records: &TransactionRecordsById,
+    ) -> Result<(), ValueTransferRecordingError> {
+        let is_received = match transaction_records.transaction_is_received(transaction_record) {
+            Ok(received) => received,
+            Err(fee_error) => {
+                return Err(ValueTransferRecordingError::FeeCalculationError(
+                    fee_error.to_string(),
+                ))
+            }
+        };
+
         let (block_height, datetime, price, pending) = (
-            transaction_md.status.get_height(),
-            transaction_md.datetime,
-            transaction_md.price,
-            !transaction_md.status.is_confirmed(),
+            transaction_record.status.get_height(),
+            transaction_record.datetime,
+            transaction_record.price,
+            !transaction_record.status.is_confirmed(),
         );
-        match (
-            transaction_md.is_outgoing_transaction(),
-            transaction_md.is_incoming_transaction(),
-        ) {
-            // This transaction is entirely composed of what we consider
-            // to be 'change'. We just make a Fee transfer and move on
-            (false, false) => (),
-            // All received funds were change, this is a normal send
-            (true, false) => {
-                for OutgoingTxData {
-                    recipient_address,
-                    value,
-                    memo,
-                    recipient_ua,
-                } in &transaction_md.outgoing_tx_data
-                {
-                    if let Ok(recipient_address) = ZcashAddress::try_from_encoded(
-                        recipient_ua.as_ref().unwrap_or(recipient_address),
-                    ) {
-                        let memos = if let Memo::Text(textmemo) = memo {
-                            vec![textmemo.clone()]
-                        } else {
-                            vec![]
-                        };
-                        summaries.push(ValueTransfer {
-                            block_height,
-                            datetime,
-                            kind: ValueTransferKind::Sent {
-                                recipient_address,
-                                amount: *value,
-                            },
-                            memos,
-                            price,
-                            txid,
-                            pending,
-                        });
-                    }
-                }
+        if is_received {
+            if !transaction_record.outgoing_tx_data.is_empty() {
+                return Err(ValueTransferRecordingError::IncoherentOutgoing(
+                    "transaction_record.outgoing_tx_data".to_string(), // TODO: Make into real data.
+                ));
```

Comment:
if you add the presence of outgoing_tx_data to the `is_received` logic we would successfully add a `Sent` value transfer even when the inputs have not been synced yet

Reply:
Good point!   I hadn't thougth carefully about the out-of-order case here.